pkg/proxy/handlers.go

```diff
@@ -10,13 +10,19 @@ import (
 	"k8s.io/client-go/transport"
 	"k8s.io/klog"
 
+	"github.com/jetstack/kube-oidc-proxy/pkg/proxy/audit"
 	"github.com/jetstack/kube-oidc-proxy/pkg/proxy/context"
 )
 
 func (p *Proxy) withHandlers(handler http.Handler) http.Handler {
 	// Set up proxy handlers
+	handler = p.auditor.WithRequest(handler)
 	handler = p.withImpersonateRequest(handler)
 	handler = p.withAuthenticateRequest(handler)
+
+	// Add the auditor backend as a shutdown hook
+	p.hooks.AddPreShutdownHook("AuditBackend", p.auditor.Shutdown)
```

Comment:
no PreStart hooks are required? I guess this hook is for flushing the backend?

Reply:
Yes, no pre start hooks are required. Didn't bother creating the logic for that reason.